protocols/rendezvous/src/client.rs

```diff
@@ -218,9 +224,18 @@ impl NetworkBehaviour for Behaviour {
     }
 
     fn on_swarm_event(&mut self, event: FromSwarm<Self::ConnectionHandler>) {
-        self.external_addresses.on_swarm_event(&event);
+        let changed = self.external_addresses.on_swarm_event(&event);
 
         self.inner.on_swarm_event(event);
+
+        if changed && self.external_addresses.iter().count() > 0 {
+            let registered = self.registered_namespaces.clone();
+            for ((rz_node, ns), ttl) in registered {
+                if let Err(e) = self.register(ns, rz_node, Some(ttl)) {
+                    log::error!("Error refreshing registration: {e}")
```

Comment:
This can only fail if we don't have addresses (which we check for) or if signing fails (which I think should basically never fail?).

Let's downgrade this to a `warn`.